### PR TITLE
1.0-RC1 Release 

### PR DESCRIPTION
**Pull request by kienstra**
* Bump versions in `amp.php` 
* Update Changelog with [PRs merged since the beta4 release](https://github.com/Automattic/amp-wp/pulls?utf8=%E2%9C%93&q=is%3Apr+is%3Aclosed+merged%3A2018-09-24T12%3A05%3A49Z..2018-10-03T03%3A17%3A43Z+milestone%3Av1.0+) (thanks to Weston for the tip in how to query by merged dates)
* Mainly copies #1457, but for `1.0-RC1`

### CODE REVIEW

File: amp.php [hunk 1]
```diff
@@ -5,7 +5,7 @@
  * Plugin URI: https://github.com/automattic/amp-wp
  * Author: WordPress.com VIP, XWP, Google, and contributors
  * Author URI: https://github.com/Automattic/amp-wp/graphs/contributors
- * Version: 1.0-beta4
+ * Version: 1.0-rc1
```
Comment (westonruter) [inline amp.php]: Let's name it `1.0-RC1` following WP conventions.

Comment (kienstra) [reply amp.php]: Sure, https://github.com/Automattic/amp-wp/pull/1482/commits/17f9eaa8a9d3f1f394d3a143adfe330a0375000a changes is to `1.0-RC1`  both above and in:
```php
define( 'AMP__VERSION', '1.0-RC1' );
```

**Comment by kienstra** [comment]
Merged in `1.0` to allow building and smoke-testing from this branch, as this is ideally the last PR of `1.0-RC1`

**Comment by kienstra** [comment]
In preparation for the `RC1` release, here's the plugin, built from this branch: [amp.zip](https://github.com/Automattic/amp-wp/files/2443383/amp.zip)

Smoke testing of it so far looks good.

**Comment by kienstra** [comment]
Here's the [amp.zip](https://github.com/Automattic/amp-wp/files/2444591/amp.zip) file, built from this branch

**Comment by kienstra** [comment]
**Smoke Testing Looks Good So Far**

Hi @westonruter,
Smoke testing of the built `amp.zip` file [above](https://github.com/Automattic/amp-wp/pull/1482#issuecomment-426856530) on the 3 staging sites has been good so far.

When you have a chance, could you please review this? Feel free to change the order of PRs in the changelog, of course.

Thanks!

**Comment by kienstra** [comment]
Here's the built [amp.zip](https://github.com/Automattic/amp-wp/files/2448812/amp.zip) file from this branch up to the last commit, https://github.com/Automattic/amp-wp/pull/1482/commits/03809b9794521eef80e6cc62560f88df2842de68.

**Comment by kienstra** [comment]
So far, the [built amp.zip](https://github.com/Automattic/amp-wp/pull/1482#issuecomment-427232954) looks good on the 3 staging sites.

**Pull request merged**